internal/controllers/clusterextension_controller.go

```diff
@@ -206,19 +206,24 @@ func (r *ClusterExtensionReconciler) reconcile(ctx context.Context, ext *ocv1alp
 	installedBundle, err := r.InstalledBundleGetter.GetInstalledBundle(ctx, ext)
 	if err != nil {
 		setInstallStatus(ext, nil)
-		// TODO: use Installed=Unknown
-		setInstalledStatusConditionFailed(ext, err.Error())
-		setStatusProgressing(ext, err)
+		// The error is put into Progressing
+		setInstalledStatusConditionUnknown(ext, err.Error())
+		setStatusProgressing(ext, errors.New("retrying to get installed bundle"))
 		return ctrl.Result{}, err
 	}
 
 	// run resolution
 	l.Info("resolving bundle")
-	resolvedBundle, resolvedBundleVersion, resolvedDeprecation, err := r.Resolver.Resolve(ctx, ext, installedBundle)
+	var bm *ocv1alpha1.BundleMetadata
+	if installedBundle != nil {
+		bm = &installedBundle.BundleMetadata
+	}
+	resolvedBundle, resolvedBundleVersion, resolvedDeprecation, err := r.Resolver.Resolve(ctx, ext, bm)
```

Comment:
Any particular reason for this change? is `nil` an unacceptable value in the resolver?

Reply:
`nil` is acceptable, and was possible if no installed bundle was found. We need to replicate that.

Reply:
To clarify, the type has changed, and can no longer be input into `Resolve()` directly. Since `installedBundle` could have been `nil`, we want to be be able to replicate that. If `Resolve()` is modified to accept the `InstalledBundle` type, this conversion would be unnecessary.